src/schema/v2/partnerArtworksMutation.ts

```diff
@@ -0,0 +1,136 @@
+interface Input {
+  id: string
+  artsyShippingDomestic: boolean | null
+  artsyShippingInternational: boolean | null
+  location: string | null
+}
+
+const PartnerArtworksType = new GraphQLObjectType<any, ResolverContext>({
+  name: "PartnerArtworks",
+  fields: () => ({
+    ...InternalIDFields,
+    success: { type: GraphQLInt },
+    errors: {
+      type: new GraphQLObjectType({
+        name: "PartnerArtworksError",
+        fields: {
+          count: { type: GraphQLInt },
+          ids: { type: GraphQLList(GraphQLString) },
+        },
+      }),
+    },
+  }),
+})
+
+const UpdatePartnerArtworksMutationSuccessType = new GraphQLObjectType<
+  any,
+  ResolverContext
+>({
+  name: "UpdatePartnerArtworksMutationSuccess",
+  isTypeOf: (data) => data.success || data.errors,
+  fields: () => ({
+    partnerArtworks: {
+      type: PartnerArtworksType,
+      resolve: (partnerArtworks) => partnerArtworks,
+    },
+  }),
+})
+
+const UpdatePartnerArtworksMutationFailureType = new GraphQLObjectType<
+  any,
+  ResolverContext
+>({
+  name: "UpdatePartnerArtworksMutationFailure",
+  isTypeOf: (data) => {
+    return data._type === "GravityMutationError"
+  },
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const UpdatePartnerArtworksMutationType = new GraphQLUnionType({
+  name: "UpdatePartnerArtworksMutationType",
+  types: [
+    UpdatePartnerArtworksMutationSuccessType,
+    UpdatePartnerArtworksMutationFailureType,
+  ],
+})
+
+export const updatePartnerArtworksMutation = mutationWithClientMutationId<
+  Input,
+  any | null,
+  ResolverContext
+>({
+  name: "UpdatePartnerArtworksMutation",
+  description: "Update all artworks that belong to the partner",
+  inputFields: {
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "ID of the partner",
+    },
+    artsyShippingDomestic: {
+      type: GraphQLBoolean,
+      description: "Whether Artsy domestic shipping should be enabled",
+    },
+    artsyShippingInternational: {
+      type: GraphQLBoolean,
+      description: "Whether Artsy international shipping should be enabled",
+    },
+    location: {
+      type: GraphQLString,
+      description: "The partner location ID to assign",
+    },
+  },
+  outputFields: {
+    partnerArtworksOrError: {
+      type: UpdatePartnerArtworksMutationType,
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: (
+    { id, artsyShippingDomestic, artsyShippingInternational, location },
+    { updatePartnerArtworksLoader }
+  ) => {
+    if (!updatePartnerArtworksMutation) {
+      throw new Error(
+        "You need to be signed in as an admin to perform this action"
+      )
+    }
+
+    const gravityOptions = {
+      artsy_shipping_domestic: artsyShippingDomestic,
+      artsy_shipping_international: artsyShippingInternational,
+      location,
+    }
+    return updatePartnerArtworksLoader?.(id, gravityOptions)
```

Comment:
By checking the loader above, can get rid of the `?`. Also, it's preferable to use `async` / `await` instead of `.then` structure.

Reply:
Updated to use `async` / `await`

Reply:
I can't seem to figure out why this can potentially be `null`. Using the [createAccountRequestMutation](https://github.com/artsy/metaphysics/blob/d7bdf8300427c70aec30b8c55737770e5ebcb277/src/schema/v2/createAccountRequestMutation.ts#L88-L124) as an example, I can't seem to find the difference.

Reply:
Ah, commented above, but:

> MP doesn't verify tokens, but it does use the existence of them being passed as headers to decide if the authenticated loaders (and typically most mutations) have loaders made available in the context.

So someone issuing this mutation without providing an `X-Access-Token` header, this loader would be `null`. That happens in: https://github.com/artsy/metaphysics/blob/d7bdf8300427c70aec30b8c55737770e5ebcb277/src/lib/loaders/index.ts#L82-L83

Reply:
It all makes sense now. Thanks!